apkid/rules/elf/obfuscators.yara

```diff
@@ -187,3 +196,24 @@ rule firehash : obfuscator
   condition:
     elf.machine == elf.EM_ARM and all of them
 }
+
+rule avdobfuscator : obfuscator
+{
+  meta:
+    description = "AVDobfuscator"
+    url         = "https://github.com/andrivet/ADVobfuscator"
+    author      = "Eduardo Novella"
+    example     = "357f0c2ad6bf5cf60c671b090eab134251db63993f52aef512bde5bfa4a1b598"
+
+  strings:
+    $o1 = "ObfuscatedAddress"
+    $o2 = "ObfuscatedCall"
+    $o3 = "ObfuscatedCallP"
+    $o4 = "ObfuscatedCallRet"
+    $o5 = "ObfuscatedCallRetP"
+    $o6 = "ObfuscatedFunc"
+
+  condition:
+    1 of ($o*) and is_elf
```

Comment:
Could this number be higher than 1?

Reply:
From what I remember NO. Many samples use a weak configuration file where only 1 routine is used.

Reply:
Hi,

Any progress so far? The rule using function's signatures seems to be working okay too. Detections at https://koodous.com/my_rulesets/4460


The rule is as follows:
```java
rule avdobfuscator : obfuscator
{
  meta:
    description = "AVDobfuscator (string signatures)"

  strings:
    $s_01 = "_ZNK17ObfuscatedAddressIPFiiiPciS0_S0_EE8originalEv"
    $s_02 = "_ZNK17ObfuscatedAddressIPFiPcEE8originalEv"
    $s_03 = "_ZNK17ObfuscatedAddressIPFvPciEE8originalEv"
    $s_04 = "_ZNK17ObfuscatedAddressIPFvPcS0_EE8originalEv"
    $s_05 = "_ZNK17ObfuscatedAddressIPFvvEE8originalEv"
    $s_06 = "_Z14ObfuscatedCallI17ObfuscatedAddressIPFvvEEJEEvT_DpOT0_"
    $s_07 = "_ZNK17ObfuscatedAddressIPFiPviEE8originalEv"
    $s_08 = "_ZNK17ObfuscatedAddressIPFvPcEE8originalEv"
    $s_09 = "_ZNK17ObfuscatedAddressIPFvP7_JNIEnvEE8originalEv"
    $s_10 = "_ZNK17ObfuscatedAddressIPFvPcS0_iiEE8originalEv"
    $s_11 = "_ZNK17ObfuscatedAddressIPFvcEE8originalEv"
    $s_12 = "_ZNK17ObfuscatedAddressIPFvPviiEE8originalEv"

  condition:
    any of them
}
```
@CalebFenton , which rule do you want to merge eventually?

Cheers

Reply:
I much prefer this second rule with `_ZNK17` prefixes.